httpgrpc/server/server.go

```diff
@@ -37,16 +38,35 @@ func NewServer(handler http.Handler) *Server {
 	}
 }
 
+type nopCloser struct {
+	*bytes.Buffer
+}
+
+func (nopCloser) Close() error { return nil }
+
+func nopCloserBuffer(b *bytes.Buffer) io.ReadCloser {
+	return nopCloser{b}
+}
+
+// BodyFromReader attempt to return the body as a bytes buffer to avoid reading and copying one that already exists.
+func BodyFromReader(r io.ReadCloser) (*bytes.Buffer, bool) {
```

Comment:
I think the idiomatic Go way to do this would be as a method, which the caller can test via type assertion.